core/src/layers/dtrace.rs

```diff
@@ -0,0 +1,227 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use async_trait::async_trait;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+
+use crate::raw::*;
+use crate::*;
+use std::ffi::CString;
+
+/// Support User Statically-Defined Tracing(aka USDT) on Linux
```

Comment:
Greate idea! Just curious, Does this USDT not only support Systemtap but also support eBPF or perf-related tools?

Reply:
Yes, you can see the example I used in PR discussion 

Actually, eBPF is more easy than systemtap in most circumstances

Reply:
So eBPF can also hook into the probe we added in this layer? Can you provide an example?

Reply:
the bpftrace is based on the eBPF which is I used in the PR description.

Reply:
Got it, thanks!